lib/Controller/ApiController.php

```diff
@@ -93,10 +93,14 @@ public function setDownloadLimit(string $token, int $limit): Response {
 
 		// Update DB
 		$shareLimit->setLimit($limit);
-		if ($insert) {
-			$this->mapper->insert($shareLimit);
+		if($limit === 0){
+			$this->mapper->delete($shareLimit);
 		} else {
-			$this->mapper->update($shareLimit);
+			if ($insert) {
```

Comment:
Should be `elseif`

Reply:
resolved @ https://github.com/nextcloud/files_downloadlimit/pull/69/commits/e14e219c96f49b3fdd27b5b93d6eee75add419cd